src/Symfony/Component/DependencyInjection/Compiler/RegisterEnvProvidersPass.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\EnvProviderInterface;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Creates the container.env_providers_locator service.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class RegisterEnvProvidersPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        $providers = array();
+        foreach ($container->findTaggedServiceIds('container.env_provider') as $id => $tags) {
+            foreach ($tags as $attr) {
+                if (isset($attr['prefix'])) {
+                    $providers[$attr['prefix']] = new Reference($id);
+                } elseif (!$r = $container->getReflectionClass($class = $container->getDefinition($id)->getClass())) {
+                    throw new InvalidArgumentException(sprintf('Class "%s" used for service "%s" cannot be found.', $class, $id));
```

Comment:
Yes, as done commonly in other passes: when it's not required at compile time, we usually don't validate that kind of things.